components/nexus-repository/src/main/java/org/sonatype/nexus/repository/view/payloads/StreamPayload.java

```diff
@@ -70,4 +71,20 @@ public String toString() {
         ", contentType='" + contentType + '\'' +
         '}';
   }
+
+  //
+  // Supplier
+  //
+
+  // NOTE: Not using Guava Supplier, as we need to express throws IOException
+
+  /**
+   * Supplies an {@link InputStream}.
+   */
+  public static interface InputStreamSupplier
+  {
+    @NotNull
+    @Nonnull
```

Comment:
Any particular reason we need both of these? Can we standardise on one annotation?